pine64/pinephone-pro/kernel/default.nix

```diff
@@ -0,0 +1,53 @@
+{
+  lib,
+  fetchFromGitLab,
+  fetchurl,
+  kernelPatches,
+  callPackage,
+  buildLinux,
+  ...
+}@args:
+let
+  pver = "6.6.44";
+
+  src_pine64 = fetchFromGitLab {
+    owner = "pine64-org";
+    repo = "linux";
+    rev = "ppp-6.6-20231104-22589";
+    sha256 = "sha256-wz2g+wE1DmhQQoldeiWEju3PaxSTIcqLSwamjzry+nc=";
+  };
+  #apply mainline fixver patches
+  upstream_patch = fetchurl {
+    url = "https://cdn.kernel.org/pub/linux/kernel/v6.x/patch-${pver}.xz";
+    hash = "sha256-SSDqB/mqUmEWcf7fhZ3RRvGc2wvauORBuVl2ovJjX5M=";
+  };
+
+  ksrc = import ./source.nix;
+
+  apply_patch = (
+    path: {
+      name = builtins.baseNameOf path;
+      patch = path;
+    }
+  );
+  patches = [ ./pinephonepro_defconfig.patch ];
```

Comment:
Is there any documentation how this patch was generated and how it can be updated?

Reply:
It was generated using `make savedefconfig` and then copied to the correct location to crate the patch by `git diff`.  But it currently is a dirty hack as i would ideally like to generated it from a nix expression as it only serves to slim the kernel down enough to make it and the initrd fit into (the first 1GiB i think) of memory. Would passing a overridden [common-config.nix](https://github.com/NixOS/nixpkgs/blob/a194f9d0654e368fb900830a19396f9d7792647a/pkgs/os-specific/linux/kernel/common-config.nix) be suitable to do so?